lib/token-persistence.spec.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+
+const fs = require('fs')
+const tmp = require('tmp')
+const readFile = require('fs-readfile-promise')
+const sinon = require('sinon')
+const { sleep } = require('wait-promise')
+const { expect } = require('chai')
+const TokenPersistence = require('./token-persistence')
+const githubAuth = require('./github-auth')
+
+describe('Token persistence', function() {
+  // Fake timers must be set up before any timers are scheduled.
+  let clock
+  beforeEach(function() {
+    clock = sinon.useFakeTimers()
+  })
+  afterEach(function() {
+    clock.restore()
+  })
+
+  beforeEach(githubAuth.removeAllTokens)
+  afterEach(githubAuth.removeAllTokens)
+
+  let path, persistence
+  beforeEach(function() {
+    path = tmp.tmpNameSync()
```

Comment:
I'm guessing tmp will automatically clean up on exit?

Reply:
It does say the following in the [options](https://www.npmjs.com/package/tmp#options) section: "keep: signals that the temporary file or directory should not be deleted on exit, default is false". I'm generally not a fan of tests leaving droppings behind them, but we're probably fine here according to the documentation! 👍